### PR TITLE
Upgrade operator sdk to 1.0

### PR DESCRIPTION
**Pull request by harche**
Signed-off-by: Harshal Patil <harpatil@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,15 +8,23 @@ An operator to perform lifecycle management (install/upgrade/uninstall) of [Kata
...
+
+   ```
+   git clone https://github.com/openshift/kata-operator 
+   git checkout -b release-4.7 --track origin/release-4.7
+   ```
+3. Install the Kata Operator on the cluster,
 
    ```
-   curl https://raw.githubusercontent.com/openshift/kata-operator/master/deploy/deploy.sh | bash
+   make install && make install && make deploy IMG=quay.io/harpatil/kata-operator:0.4
```
Comment (harche) [inline README.md:21]: Keeping it pinned to a personal image until we have the image deployed in official repository

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+resources:
+- manager.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+images:
+- name: controller
+  newName: quay.io/harpatil/kata-operator
```
Comment (harche) [inline config/manager/kustomization.yaml:7]: This will change when we have official image.

File: controllers/openshift_controller.go [hunk 3]
```diff
@@ -129,12 +149,13 @@ func (r *ReconcileKataConfigOpenShift) processDaemonsetForCR(operation DaemonOpe
...
-					ServiceAccountName: "kata-operator",
+					ServiceAccountName: "default",
 					NodeSelector:       nodeSelector,
 					Containers: []corev1.Container{
 						{
-							Name:            "kata-install-pod",
-							Image:           "quay.io/isolatedcontainers/kata-operator-daemon:4.6",
+							Name: "kata-install-pod",
+							// Image:           "quay.io/isolatedcontainers/kata-operator-daemon:v1.0",
+							Image:           "docker.io/pharshal/kata-install-daemon:0.1.1",
```
Comment (harche) [inline controllers/openshift_controller.go:158]: This will change when we have official image for daemonset built

**Review by jensfr — APPROVED** [review]
It's a whole lot of new code. Some name changes like e.g. for the service account name etc surprised me but I think that's because more of it is now generated by operator-sdk. So I think we should merge this and then fix everything we need to.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,123 @@
+# Current Operator version
+VERSION ?= 0.0.1
```
[the remark this answers is not in the document]
Comment (harche) [reply Makefile]: done. set it to `4.7.0`

**Pull request merged**